controllers/useraccount/console_usersettings.go

```diff
@@ -0,0 +1,35 @@
+package useraccount
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ConsoleUserSettingsIdentifier         = "console.openshift.io/user-settings"
+	ConsoleUserSettingsUID                = "console.openshift.io/user-settings-uid"
+	UserSettingNS                         = "openshift-console-user-settings"
+	ConsoleUserSettingsResourceNamePrefix = "user-settings-"
+)
+
+// deleteResource deletes the specified resource associated with a user from console setting.
+// It first attempts to delete the resource by name, and if not found, it deletes all resources with matching labels.
+//
+// userUID : The unique identifier of the user for whom the resource is being deleted.
+// Returns an error if the deletion operation fails, or nil if the operation is successful.
+func deleteResource(ctx context.Context, cl client.Client, userUID string, toDelete client.Object) error {
+
+	name := ConsoleUserSettingsResourceNamePrefix + userUID
+	toDelete.SetName(name)
+	toDelete.SetNamespace(UserSettingNS)
+	if err := cl.Delete(ctx, toDelete); err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
out of curiosity? is there any reason for deleting via a label only if the resource with the given name is not found? I mean, why couldn't we simply run both delete commands (by name and by a label) for every user that is being de-provisioned?

Reply:
In the ideal scenario it should be fine to just delete resources via label but since the label will only be present on newer objects ( once https://issues.redhat.com//browse/OCPBUGS-32321 gets implemented) - this function looks for the objects by name. 
The reason why it looks for deleting via label only after the resource name is not found by name is because it would be redundant trying to delete the object twice (since ideally there should not be more than one cm, role and rb for a user)